native/cocos/platform/android/java/vendor/google/billing/GoogleBillingHelper.java

```diff
@@ -24,191 +24,512 @@ of this software and associated documentation files (the "Software"), to deal
+    }
+
+    public static final class BillingClientPurchasesUpdatedListener implements PurchasesUpdatedListener {
+        private int _tag;
+        BillingClientPurchasesUpdatedListener(int tag) {
+            this._tag = tag;
+        }
+        @Override
+        public void onPurchasesUpdated(@NonNull BillingResult billingResult, @Nullable List<Purchase> purchaseList) {
+            int tag = this._tag;
+            CocosHelper.runOnGameThread(new Runnable() {
+                @Override
+                public void run() {
+                    GoogleBilling billing = googleBillings.get(tag);
+                    if (billing != null) {
+                        int startID = billing.pushPurchases(purchaseList);
+                        GoogleBillingHelper.onPurchasesUpdated(tag, billingResult, purchaseList, startID);
+                    }
+                }
+            });
+        }
+    }
+
+    public static final class BillingClientUserChoiceBillingListener implements UserChoiceBillingListener {
+        private int _tag;
+        BillingClientUserChoiceBillingListener(int tag) {
+            this._tag = tag;
+        }
+        @Override
+        public void userSelectedAlternativeBilling(@NonNull UserChoiceDetails userChoiceDetails) {
+            int tag = this._tag;
+            CocosHelper.runOnGameThread(new Runnable() {
+                @Override
+                public void run() {
+                    GoogleBilling billing = googleBillings.get(tag);
+                    if (billing != null) {
+                        GoogleBillingHelper.userSelectedAlternativeBilling(tag, userChoiceDetails);
+                    }
+                }
+            });
+        }
+    }
+
+    private static native void onBillingSetupFinished(int tag, int callbackId, @NonNull BillingResult billingResult);
+    private static native void onBillingServiceDisconnected(int tag, int callbackId);
+    private static native void onProductDetailsResponse(int tag, int callbackId, @NonNull BillingResult billingResult, @NonNull List<ProductDetails> productDetailsList, int startID);
+    private static native void onPurchasesUpdated(int tag, @NonNull BillingResult billingResult, @Nullable List<Purchase> purchasesList, int startID);
+    private static native void userSelectedAlternativeBilling(int tag, @NonNull UserChoiceDetails userChoiceDetails);
+    private static native void onQueryPurchasesResponse(int tag, int callbackId, @NonNull BillingResult billingResult, @NonNull List<Purchase> purchasesList, int startID);
+    private static native void onConsumeResponse(int tag, int callbackId, @NonNull BillingResult billingResult, @NonNull String purchaseToken);
+    private static native void onAcknowledgePurchaseResponse(int tag, int callbackId, @NonNull BillingResult billingResult);
+    private static native void onBillingConfigResponse(int tag, int callbackId, @NonNull BillingResult billingResult, @Nullable BillingConfig billingConfig);
```

Comment:
Please check whether the function will be obfuscated. If it will be obfuscated, the corresponding configuration needs to be added to proguard-rules.pro

Reply:
fixed.